From 1dba7da22bed0dcf8e06fb225141049e46d67665 Mon Sep 17 00:00:00 2001
From: Christoph Reiter <reiter.christoph@gmail.com>
Date: Fri, 23 Jul 2021 08:52:50 +0200
Subject: [PATCH 102/N] time: fix strftime not raising for invalid year
 values

It's crt specific and not compiler specific.
This fixes a test case in test_time

Signed-off-by: Naveen M K <naveen521kk@gmail.com>
---
 Modules/timemodule.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Modules/timemodule.c b/Modules/timemodule.c
index 4caacc3..4787fa3 100644
--- a/Modules/timemodule.c
+++ b/Modules/timemodule.c
@@ -759,7 +759,7 @@ time_strftime(PyObject *self, PyObject *args)
         return NULL;
     }
 
-#if defined(_MSC_VER) || (defined(__sun) && defined(__SVR4)) || defined(_AIX) || defined(__VXWORKS__)
+#if defined(MS_WINDOWS) || (defined(__sun) && defined(__SVR4)) || defined(_AIX) || defined(__VXWORKS__)
     if (buf.tm_year + 1900 < 1 || 9999 < buf.tm_year + 1900) {
         PyErr_SetString(PyExc_ValueError,
                         "strftime() requires year in [1; 9999]");
-- 
2.34.1

